id_ID/ch2-id.md

```diff
@@ -23,30 +23,30 @@ Kami berharap Little Carrot memiliki kemampuan **bergerak secara otonom**. Meski
 
 Pertama-tama, sebuah robot membutuhkan roda dan motor untuk bergerak, sehingga kita diinstal roda di bawah Little Carrot (gait kontrol untuk robot-robot humanoid sangat rumit, yang kita tidak akan mempertimbangkannya di sini). Sekarang dengan roda, robot dapat bergerak, tetapi tanpa sistem kontrol yang efektif, Little Carrot tidak tahu mana target manay yang melakukan tindakan, dan itu bisa melakukan apa-apa tapi berkeliaran membabi buta. Lebih buruk lagi, mungkin memukul dinding dan menyebabkan kerusakan. Untuk menghindari hal terjadi, kami memasang kamera di atas kepalanya, dengan intuisi bahwa robot **harus terlihat mirip manusia** - kita bisa melihatnya dari sketsa. Dengan mata, otak dan tungkai, manusia dapat berjalan bebas dan menjelajahi lingkungan apapun, jadi kami (mungkin naif) berpikir bahwa robot kami harus mampu mencapainya. Untuk membuat Little Carrot dapat menjelajahi ruang, dibutuhkan untuk mengetahui setidaknya dua hal:
 
-1. Where am I? - positioning
-2. what is the surrounding environment like? - map building
+1. Dimana saya? - penempatan
+2. seperti apa lingkungan di sekitarnya? - pembangunan peta
 
-"Positioning" and "map building", can be seen as perception in both directions: inward and outward. As a well-rounded robot, Little Carrot need not only to understand its own **state** (i.e. location), but also the external **environment** (i.e. map). Of course, there are many ways to solve these two problems. For example, we can lay guiding rails on the floor of the room, or stick markers such as pictures containing QR code on the wall, or mount radio positioning devices on the table. If you are outdoor, you can also install a positioning component (like the one in a cell phone or a car) on the head of Little Carrot. With these devices, can we claim that the positioning problem has been resolved? Let's categorize these sensors (see \ autoref {fig: sensors}) into two classes.
+"Penempatan" dan "pembangunan peta", dapat dilihat sebagai persepsi dua arah: ke dalam dan ke luar. Sebagai robot yang baik, _Little Carrot_ tidak perlu memahami **negara** (lokasinya) sendiri, tetapi juga lingkungan eksternalnya (peta). Tentu saja, ada banyak cara untuk mengatasi kedua masalah ini. Misalnya, kita bisa meletakkan rel pemandu di lantai ruangan, atau spidol stick seperti gambar yang berisi kode QR di dinding, atau memasang perangkat posisi radio di atas meja. Jika Anda berada di luar ruangan, Anda juga dapat memasang komponen penempatan (seperti yang ada pada ponsel atau mobil) di kepala _Little Carrot_. Dengan perangkat ini, dapatkah kita mengklaim bahwa masalah penentuan penempatan sudah terpecahkan? Mari kita kategorikan sensor ini (see \ autoref {fig: sensors}) menjadi dua kelas.
```

Comment:
- 'state' should not be translated to negara, which means country.
- ... **environment** .. in the original text is not being translated with constant formatting.